File: src-kotlin/task-manager-domain/src/main/kotlin/com/example/taskmanager/domain/tasks/TaskStatus.kt
package com.example.taskmanager.domain.tasks

/**
 * Lifecycle states for a workshop task.
 */
enum class TaskStatus {
    TODO,
    IN_PROGRESS,
    DONE,
    CANCELLED,
    ;

    val isActive: Boolean
        get() = this != DONE && this != CANCELLED
}

File: src-swift/Sources/TaskManagerDomain/TaskStatus.swift
/// Lifecycle states for a workshop task.
public enum TaskStatus: String, Codable, CaseIterable, Sendable {
    case todo = "TODO"
    case inProgress = "IN_PROGRESS"
    case done = "DONE"
    case cancelled = "CANCELLED"

    public var isActive: Bool {
        self != .done && self != .cancelled
    }

    public var displayName: String {
        switch self {
        case .todo:
            return "To Do"
        case .inProgress:
            return "In Progress"
        case .done:
            return "Done"
        case .cancelled:
            return "Cancelled"
        }
    }
}
